src/node/txdownloadman_impl.cpp

```diff
@@ -444,4 +444,58 @@ void TxDownloadManagerImpl::MempoolRejectedPackage(const Package& package)
 {
     RecentRejectsReconsiderableFilter().insert(GetPackageHash(package));
 }
+
+std::pair<bool, std::optional<PackageToValidate>> TxDownloadManagerImpl::ReceivedTx(NodeId nodeid, const CTransactionRef& ptx)
+{
+    const uint256& txid = ptx->GetHash();
+    const uint256& wtxid = ptx->GetWitnessHash();
+
+    // Mark that we have received a response
+    m_txrequest.ReceivedResponse(nodeid, ptx->GetHash());
+    if (ptx->HasWitness()) m_txrequest.ReceivedResponse(nodeid, ptx->GetWitnessHash());
```

Comment:
In f3cc6322798ca61a31f90b2ad7a0c32471db0410 "[refactor] move new tx logic to txdownload"

nit:

```suggestion
    m_txrequest.ReceivedResponse(nodeid, txid);
    if (ptx->HasWitness()) m_txrequest.ReceivedResponse(nodeid, wtxid);
```

Reply:
done